ouroboros-network/src/Ouroboros/Network/Mux/Types.hs

```diff
@@ -145,6 +190,9 @@ newtype MiniProtocolDispatch ptcl m =
 data MiniProtocolMode = ModeInitiator | ModeResponder
   deriving (Eq, Ord, Ix, Enum, Bounded, Show)
 
+data MuxStyle = StyleClient | StyleServer
```

Comment:
@karknu could you describe here what's the purpose of `MuxStyle`, is it something like `PeerKind` in `typed-protocols`?

Reply:
`PeerKind` seems to be missing from the repo at the moment.
It is used to decide if the Mux bearer should initiate version negotiation at once or wait for a peer to contact it.

Reply:
So the client of our `ouroboros-network` library will need to decided what they need. If it's just a simple wallet node which connects to a single (or multiple) relays, they will need to run just a client of this protocol.  But a relay node or a core node, will need to run both client (to be able to connect at startup to some nodes) and a server to be able to respond to incoming connections.

Reply:
Until peer discovery is implemented a simple wallet node could get away with only running as a client. But the intention was that an `ouroboros-network` library user would always start in server mode and then as a client for as many peers as it liked. I guess `node-to-wallet` interface changed this slightly since one side is always client and the other side always server in that case.